src/main/scala/com/wavesplatform/state2/VersionableStorage.scala

```diff
@@ -0,0 +1,20 @@
+package com.wavesplatform.state2
+
+trait VersionableStorage {
+  this: VariablesStorage =>
+
+  protected val Version: Option[Int]
+  private val stateVersion = "stateVersion"
+
+  def isVersionValid: Boolean =
```

Comment:
What is the point of making it a part of class and all the ineritance?
make it a function and pass it as predicate.